sdcm/utils/issues.py

```diff
@@ -28,6 +30,26 @@ class Issue:
     issue_id: int | None
 
 
+class CachedGitHubIssues:
+    def __init__(self):
+        self.storage = KeyStore()
+
+    @lru_cache()
+    def get_repo_data(self, owner, repo):
+        scsv = self.storage.get_file_contents(f'issues/{owner}_{repo}.csv')
+        print('got repo')
+        return {issue['id']: issue for issue in csv.DictReader(scsv.decode().splitlines(), fieldnames=("id", "state", "labels"))}
+
+    def get_issue(self, owner: str, repo_id: str, issue_id: str | int):
+        repo_issues_mapping = self.get_repo_data(owner, repo_id)
+        return repo_issues_mapping.get(str(issue_id))
+
+
+@lru_cache(maxsize=1)
+def get_issues_cache():
+    return CachedGitHubIssues()
```

Comment:
can also make it singleton class

Reply:
just used @cache on the class, it's exactly the same